### ISSUE TITLE
Remove records returned by query_data loop with Python

### ISSUE THREAD

**Issue by edigiacomo** [ARPA-SIMC/dballe#140]
In dballe8, the `context_id` is removed, so I have to change the following code:

```python
contexts_to_del = []
for rec in db.query_data(dballe.Record()):
    if rec["date"] not in steps:
        contexts_to_del.append(rec["context_id"])

for ctx in contexts_to_del:
    db.remove(dballe.Record(context_id=ctx))
```

I tried with the following code:

```python
records_to_del = []
for rec in db.query_data(dballe.Record()):
    if rec["datetime"] not in steps:
        records_to_del.append(rec)

for rec in records_to_del:
    db.remove_data(rec)
```

But it raises an exception:

```
trans.remove_data(rec)
TypeError: Expected dict or None
```

**TL;DR** what is the proper way to delete a record from a DB from a `CursorDataDB` object?

Below, my attemps:

I tried something more verbose:

```python
for rec in db.query_data({}):                                                  
    if rec["datetime"] not in steps:                                           
        records_to_del.append({k: rec[k] for k in (                            
            "ident", "lon", "lat", "rep_memo",                                 
            "leveltype1", "l1", "leveltype2", "l2",                            
            "pindicator", "p1", "p2",                                          
            "datetime"                                                         
        )})                                                                    
                                                                                   
for rec in records_to_del:                                                     
    db.remove_data(rec)
```

But the coordinates are `Decimal`:

```
db.remove_data(rec)
TypeError: longitude value must be an instance of int, float, str, or None
```

I could statically expand the code to convert the `Decimal` to `float`, but I don't know if can run into some floating point errors:

```python
for rec in db.query_data({}):                                                   
    if rec["datetime"] not in steps:                                            
        r = {                                                                   
            k: rec[k] for k in (                                                
                "ident", "rep_memo",                                            
                "leveltype1", "l1", "leveltype2", "l2",                         
                "pindicator", "p1", "p2",                                       
                "datetime"                                                      
            )                                                                   
        }                                                                       
        r["lon"] = float(rec["lon"])                                            
        r["lat"] = float(rec["lat"])                                            
        records_to_del.append(r)                                                
                                                                                
for rec in records_to_del:                                                      
    db.remove_data(rec) 
```

**Comment by edigiacomo** [comment]
The same problem occurs when trying to insert new data derived from a `DB.query_data` object: the coordinates are incompatible. Can I safely convert to `float` or to `int` (after multiplying by 10⁵)? Or maybe it's better to implement a helper function, e.g. `CursorDataDB.to_dict` and/or accept `Decimal` coordinates?

**Comment by spanezz** [comment]
Ok, so, there are various problems here. One is accessing `context_id`, which can be done with:
```
records_to_del = []
for rec in db.query_data():
    if rec["datetime"] not in steps:
        records_to_del.append(rec["context_id"])
```
However, deleting by `context_id` doesn't seem to have been supported since a long time. Let's talk about how to do it.

**Comment by spanezz** [comment]
I have committed all the fixes to this to the branch `dballe8`. Now you can call `.query` on a cursor to get a dict that can be used to query the exact value at the cursor, and Decimals are accepted for latitude and longitude

**Comment by edigiacomo** [comment]
Thanks! I'll package the new pre-release and try it! :+1:

**Comment by spanezz** [comment]
For reference, your code becomes:
```
to_del = []
for rec in db.query_data(dballe.Record()):
    if rec["date"] not in steps:
        to_del.append(cur.query)

for query in to_del:
    db.remove(query)
```
or perhaps more efficiently:
```
to_del = set()
for rec in db.query_data():
    if rec["datetime"] not in steps:
        to_del.add(rec["date"])

for query in to_del:
    db.remove({"datetime": query})
``` 

If you think it could be a good idea, feel free to open a feature request ticket for having `cursor.remove()` implemented.

**Comment by spanezz** [comment]
This was implemented in d19d9303d17abda1cf3b11d1978d5501d1d0be6d, I apparently missed updating the ticket :/
